core/shared/src/test/scala/fs2/TextSuite.scala

```diff
@@ -136,13 +139,13 @@ class TextSuite extends Fs2Suite {
     group("handles byte order mark") {
       val bom = Chunk[Byte](0xef.toByte, 0xbb.toByte, 0xbf.toByte)
       property("single chunk") {
-        forAll { (s: String) =>
+        forAll(genStringNoBom) { (s: String) =>
```

Comment:
This seems like a very ironic diff.

Reply:
@djspiewak Took me a minute to catch on.

BTW, I'm not sure FS2 should really strip the byte order mark -- if FS2 just passed it along, these tests would all have passed. Looking at https://github.com/typelevel/fs2/issues/1484, we never really discussed why removing it is right behavior. Here's a quote from wikipedia: "The standard also does not recommend removing a BOM when it is there, so that round-tripping between encodings does not lose information, and so that code that relies on it continues to work." (https://en.wikipedia.org/wiki/Byte_order_mark). Thoughts?

Reply:
IMO if the conversion is to `String`, it's better to strip the BOM. `String` itself already internally establishes all of the invariants that are communicated via the BOM (codec and endianness), so it's just redundant information which messes up offsets invisibly. However, as long as things are in byte-form, the BOM needs to be preserved because we have no other way of signalling those things.

In other words, I *think* I believe that `utf8decode` is doing the right thing. A `Pipe[F, Byte, String]` should strip the BOM, while a `Pipe[F, Byte, Byte]` (or analogous) should not.